archivist/archivistpublic.py

```diff
@@ -0,0 +1,428 @@
+
+        return newheaders
+
+    # the public endpoint is currently readonly so only read-type methods are
+    # defined here. This may change - the Public endpoint may allow writes
+    # in future...
+    @retry_429
+    def get(
+        self,
+        url: str,
+        *,
+        headers: Optional[Dict] = None,
+        params: Optional[Dict] = None,
+    ) -> Dict:
+        """GET method (REST)
+
+        Args:
+            url (str): e.g. https://app.rkvst.io/archivist/v2/publicassets/xxxxxxxxxxxxxxxxxxxxxx
+            headers (dict): optional REST headers
+            params (dict): optional params strings
+
+        Returns:
+            dict representing the response body (entity).
+
+        """
+        response = self.session.get(
+            url,
+            headers=self._add_headers(headers),
+            verify=self.verify,
+            params=_dotdict(params),
+        )
+
+        self._response_ring_buffer.appendleft(response)
+
+        error = _parse_response(response)
+        if error is not None:
+            raise error
+
+        return response.json()
+
+    @retry_429
+    def get_file(
+        self,
+        url: str,
+        fd: BinaryIO,
+        *,
+        headers: Optional[Dict] = None,
+        params: Optional[Dict] = None,
+    ) -> Response:
+        """GET method (REST) - chunked
+
+        Downloads a binary object from upstream storage.
+
+        Args:
+            url (str): e.g. assets/xxxxxxxxxxxxxxxxxxxxxx
+            identity (str): e.g. attachments/xxxxxxxxxxxxxxxxxxxxxxxxxxxx`
+            fd (file): an iterable representing a file (usually from open())
+                the file must be opened in binary mode
+            headers (dict): optional REST headers
+            params (dict): optional params strings
+
+        Returns:
+            REST response (not the response body)
+
+        """
+        response = self.session.get(
+            url,
+            headers=self._add_headers(headers),
+            verify=self.verify,
+            stream=True,
+            params=_dotdict(params),
+        )
+
+        self._response_ring_buffer.appendleft(response)
+
+        error = _parse_response(response)
+        if error is not None:
+            raise error
+
+        for chunk in response.iter_content(chunk_size=4096):
+            if chunk:
+                fd.write(chunk)
+
+        return response
+
+    @retry_429
+    def __list(
+        self,
+        url: str,
+        params: Optional[Dict],
+        *,
+        page_size: Optional[int] = None,
+        headers: Optional[Dict] = None,
+    ) -> Response:
+        if page_size is not None:
+            if params is not None:
+                params["page_size"] = page_size
+            else:
+                params = {"page_size": page_size}
+
+        response = self.session.get(
+            url,
+            headers=self._add_headers(headers),
+            verify=self.verify,
+            params=_dotdict(params),
```

Comment:
The signature of dotdict causes a liniting error with pylance. The fix is : `def _dotdict(dct: Optional[dict]) -> Optional[dict]:`

Reply:
fixed